libs/libc/tls/tls_alloc.c

```diff
@@ -57,34 +54,33 @@
 
 int tls_alloc(void)
 {
-  FAR struct tcb_s *rtcb = this_task();
-  FAR struct task_group_s *group = rtcb->group;
-  irqstate_t flags;
+  FAR struct task_info_s *tinfo = task_get_info();
   int candidate;
   int ret = -EAGAIN;
```

Comment:
Inconsistent error reporting.  Some functions call sem_wait() which sets the errno and returns ERROR on and error.  Others follow the internal OS  convention of returning a negated errno value.  This needs to be made consistent and documented.